lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/util/Procedures.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.base.util;
+
+import io.trino.spi.TrinoException;
+
+import static io.trino.spi.StandardErrorCode.INVALID_PROCEDURE_ARGUMENT;
+import static java.lang.String.format;
+
+public final class Procedures
+{
+    private Procedures() {}
+
+    public static void checkProcedureArgument(boolean condition, String message, Object... args)
```

Comment:
Add `@FormatMethod` annotation

Reply:
> I find the name distributedWithFilteringAndRepartitioning() confusing. Would supportsFiltering=false also do repartitioning? More generally, how does filtering relate to repartitioning?

I don't know, this code was here before.

Reply:
support for repartitioning is implicitly bounded to "distributed" execution mode. If a specific redistribution (non-aribtrary) happens depends on the layout returned by the connector from `getLayoutForTableExecute`.